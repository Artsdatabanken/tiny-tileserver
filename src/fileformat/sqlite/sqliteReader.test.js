const reader = require("./sqliteReader");

test("listTables", () => {
  return reader
    .listTables("./testdata/test.sqlite")
    .then(tables => expect(tables).toMatchSnapshot());
});

test("listTables missing file", () => {
  return reader
    .listTables("./testdata/missing.sqlite")
    .catch(err => expect(err).toMatchSnapshot());
});

test("getColumns", () => {
  return reader
    .getColumns("./testdata/test.sqlite", "threecolumns")
    .then(columns => expect(columns).toMatchSnapshot());
});

test("listRows", () => {
  return reader
    .listRows("./testdata/test.sqlite", "threecolumns", ["col2", "col3"])
    .then(value => expect(value).toMatchSnapshot());
});

test("read", () => {
  return reader
    .read(
      "./testdata/test.sqlite",
      "threecolumns",
      [1, "a"],
      ["col1", "col2", "col3"]
    )
    .then(value => expect(value).toMatchSnapshot());
});
test("read missing", () => {
  return reader
    .read(
      "./testdata/test.sqlite",
      "threecolumns",
      [4],
      ["col1", "col2", "col3"]
    )
    .then(value => expect(value).toMatchSnapshot());
});

test("read invalid tablename", () => {
  return reader
    .read("./testdata/test.sqlite", "missingtable", [4], ["col1"])
    .catch(err => expect(err).toMatchSnapshot());
});
