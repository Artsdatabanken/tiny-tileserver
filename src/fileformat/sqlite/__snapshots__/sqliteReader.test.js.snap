// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getColumns 1`] = `
Array [
  "col1",
  "col2",
  "col3",
]
`;

exports[`listRows 1`] = `
Array [
  Object {
    "name": "a",
    "size": 3,
  },
  Object {
    "name": "b",
    "size": 0,
  },
  Object {
    "name": "",
    "size": null,
  },
]
`;

exports[`listTables 1`] = `
Array [
  Object {
    "name": "threecolumns",
  },
]
`;

exports[`listTables missing file 1`] = `[Error: SQLITE_CANTOPEN: unable to open database file]`;

exports[`read 1`] = `"abc"`;

exports[`read invalid tablename 1`] = `[Error: SQLITE_ERROR: no such table: missingtable]`;

exports[`read missing 1`] = `null`;
